Source/WebKit/Platform/IPC/cocoa/ConnectionCocoa.mm

```diff
@@ -424,48 +425,47 @@ static void clearNoSenderNotifications(mach_port_t port)
     Vector<Attachment> attachments(numberOfAttachments);
 
     for (mach_msg_size_t i = 0; i < numberOfAttachments; ++i) {
-        mach_msg_descriptor_t* descriptor = reinterpret_cast<mach_msg_descriptor_t*>(descriptorData);
-        ASSERT(descriptor->type.type == MACH_MSG_PORT_DESCRIPTOR);
-        if (descriptor->type.type != MACH_MSG_PORT_DESCRIPTOR)
+        auto& descriptor = spanReinterpretCast<mach_msg_port_descriptor_t>(remaining.first(sizeof(mach_msg_port_descriptor_t)))[0];
```

Comment:
This idiom with `first`, `spanReinterpretCast`, and `[0]` needs a wrapper function to make it look cleaner and avoid repeating the type twice. Maybe involving the word _head_ or _front_?

Reply:
Proposing:
```
template<typename T, typename Extent>
const T& reinterpretCastSpanStartTo(std::span<const uint8_t, Extent> span)
{
    return spanReinterpretCast<const T>(span.first(sizeof(T)))[0];
}

template<typename T, typename Extent>
T& reinterpretCastSpanStartTo(std::span<uint8_t, Extent> span)
{
    return spanReinterpretCast<T>(span.first(sizeof(T)))[0];
}
```